image-download

```diff
@@ -83,8 +83,23 @@ def find(name, stores, latest, quiet):
     for store in stores:
         url = urllib.parse.urlparse(urllib.parse.urljoin(store, name))
 
-        # First, try the easy case: for OpenShift proxied stores which send
-        # their own SSL cert initially host name has to match that
+        # First, check if this is an S3 store for which we have a key
+        if s3.is_key_present(url):
+            output = check_curl_args([s3.sign_url(url, verb='HEAD')])
+            if output:
```

Comment:
Is that guaranteed to indicate success? Or could it also be a non-empty error message?

Reply:
the `check` in `subprocess.check_output()` means that non-zero returns are raised as exceptions.  In that case we return `None`.  the `--fail` to `curl` means that non-success will return non-zero exit status.

This is the same basic mechanism we relied on before, except that we did the work inside of the `try` branch.  With the reorg, the only way this function returns non-`None` is the exact case where we were inside of the `try` branch before.